### PR TITLE
Add "fadeOutAudio()" to background.js

### PR DESCRIPTION
**Pull request by AFellowSpeedrunner**
Not sure if it's entirely perfect but at least it works. Didn't like how it abruptly cuts off so I decided to add some sort of fade out code to the background.js file.

### CODE REVIEW

**Comment by AFellowSpeedrunner** [comment]
I have absolutely no clue why it re-added the other changes but I did yesterday that were already merged. Wouldn't break anything anyways.

I'm not the best at this but this should help it sound a lot more natural with the fade out when switching sites without an abrupt stop to the song.

File: scripts/background.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ setInterval(() => {
                             { src: '/images/settings-icon.png', sizes: '256x256', type: 'image/png' }
                         ]
                     });
-                };
+                }
```
Comment (WebSnke) [inline scripts/background.js:31]: May I know why you removed the semicolons?

File: scripts/background.js [hunk 2]
```diff
@@ -40,34 +40,47 @@ setInterval(() => {
...
+                fadeOut();
+            }
         });
 }, 1000);
 
+function fadeOut() {
+    let volume = audio.volume;
+    const fadeOutInterval = setInterval(() => {
+        if (volume > 0) {
+            volume -= 0.001;
```
Comment (WebSnke) [inline scripts/background.js:54]: I think subtracting by 0.001 is a bit slow and 0.01 sounds better.

```suggestion
            volume -= 0.01;
```

File: scripts/background.js [hunk 3]
```diff
@@ -76,7 +89,7 @@ browser.runtime.onInstalled.addListener((details) => {
             type: 'basic',
             iconUrl: browser.extension.getURL('/images/warning.png'),
             title: 'WARNING - HEALTH AND SAFETY',
-            message: 'BEFORE PLAYING, READ THE HEALTH AND SAFETY PRECAUTIONS BOOKLET FOR IMPORTANT INFORMATION ABOUT YOUR HEALTH AND SAFETY.',
+            message: 'BEFORE PLAYING, READ THE HEALTH AND SAFETY PRECAUTIONS BOOKLET FOR IMPORTANT INFORMATION ABOUT YOUR HEALTH AND SAFETY.'
```
Comment (WebSnke) [inline scripts/background.js:92]: Thanks. You are right.

**Review by WebSnke — CHANGES_REQUESTED** [review]

File: scripts/background.js (hunk 1 above)
Comment (AFellowSpeedrunner) [reply scripts/background.js:31]: I have no clue why the semicolons are removed. Must of been my code editor. I'll readd them soon and do the other stuff.

**Comment by AFellowSpeedrunner** [comment]
Everything should now be fixed!

File: scripts/background.js [hunk 4]
```diff
@@ -42,14 +42,27 @@ setInterval(() => {
...
                 };
             } else if (!regex.settings.test(url) && !regex.shopping.test(url) && !audio.paused && !audio.src.includes(audioSource.health)) {
-                audio.pause();
+                fadeOut();
             };
         });
 }, 1000);
 
+function fadeOut() {
+    let volume = audio.volume;
```
Comment (WebSnke) [inline scripts/background.js:51]: I suggest adding "initialVolume" to be able to set the volume to its previous value.

```js
const initialVolume = audio.volume;
const fadeOutInterval = setInterval(() => {
    if (audio.volume > 0) {
        audio.volume -= 0.01;
    } else {
        clearInterval(fadeOutInterval);
        audio.pause();
        audio.volume = initialVolume;
    }
}, 1);
```

I haven't changed these changes fully yet. Let me know if you find any issues.

**Comment by AFellowSpeedrunner** [comment]
Alright, I'm going to test it in a minute or so. I'll report back if everything works okay.

File: scripts/background.js (hunk 4 above)
Comment (WebSnke) [reply scripts/background.js:51]: > Added it in.

Apparently not. It does not show up on GitHub.

**Review by WebSnke — CHANGES_REQUESTED** [review]

**Comment by AFellowSpeedrunner** [comment]
Found an issue I had when originally adding this pop up again. It isn't fading out all the way with 0.01. I remember setting it to 0.001 fixed it. Also, now it doesn't go back up when switching tabs due to this. Going to see if something broke that bit.

File: scripts/background.js (hunk 4 above)
Comment (AFellowSpeedrunner) [reply scripts/background.js:51]: I see it on my commit list on my repo under commit bae4bf3. I was able see it fine. Try checking it again.

**Comment by AFellowSpeedrunner** [comment]
This should hopefully fix it. Reason why it wouldn't go down all the way is because it kept trying to set it negative and it really doesn't like that causing hundreds upon thousands of errors to happen within seconds. Eventually, this will cause Firefox (at least on my MacBook) to lock and cause the system to start slowing down. My code now fixes that by clamping volumes to 0 minimum.

File: scripts/background.js (hunk 4 above)
Comment (AFellowSpeedrunner) [reply scripts/background.js]: Updated it anyways so it should now definitely appear for you.

File: scripts/background.js [hunk 5]
```diff
@@ -42,14 +42,25 @@ setInterval(() => {
...
-                audio.pause();
+                fadeOutAudio();
             };
         });
 }, 1000);
 
+function fadeOutAudio() {
+    const initialVolume = audio.volume;
+    const fadeOutInterval = setInterval(() => {
+	audio.volume = Math.max(0, audio.volume - 0.01);
```
Comment (WebSnke) [inline scripts/background.js:53]: I thibnk this now sounds best.

```suggestion
	audio.volume = Math.max(0, audio.volume - 0.005);
```

File: scripts/background.js [hunk 6]
```diff
@@ -42,14 +42,25 @@ setInterval(() => {
...
+        if (audio.volume === 0) {
+            clearInterval(fadeOutInterval);
+	    audio.pause();
+            audio.volume = initialVolume;
+        }
+    }, 5);
```
Comment (WebSnke) [inline scripts/background.js:59]: Same here.

```suggestion
    }, 10);
```

**Review by WebSnke — CHANGES_REQUESTED** [review]

**Comment by AFellowSpeedrunner** [comment]
Alright, committed them over.

**Review by WebSnke — APPROVED** [review]

**Comment by AFellowSpeedrunner** [comment]
Nice, thanks for the merge!

**Pull request merged**